### ISSUE TITLE
Problem with viewport.destroy() when React component gets dismounted

### ISSUE THREAD

**Issue by darotar** [davidfig/pixi-viewport#473]
I've experienced a problem when I was destroying the viewport on component dismounting in React

the dismount logic is pretty straight-forward:
```typescript
React.useEffect(() => {
   return () => {
      viewport.destroy();
   };
})
```

The problem occurs connected with removing event listeners from domElement when destroying - as in React dismounting, markup's being removed from DOM first and side effects get done secondy, thus it throws the error that there's no "removeEventListener" of "undefined", domElement doesn't exist already

<img width="818" alt="image" src="https://github.com/davidfig/pixi-viewport/assets/15818529/d6938d50-a099-41e0-978d-d8da169e3169">

As my current workaround, I'm just adding fake domElement because I see these final steps are only related to event listeners removing and they were blocking further going to super.destroy() part

```typescript
React.useEffect(() => {
   return () => {
      let div = document.createElement('div');

      if (!viewport.options.events.domElement) {
         viewport.options.events.domElement = div;
      }

      viewport.destroy({
         children: true,
         texture: true,
         baseTexture: true,
      });

      div = undefined;
   }
});
```

I don't consider it a viewport library issue, as viewport's flow just wants to make things in right order, just was wondering if there's still some adjustments required that would allow to avoid such issues.

E.g. I could raise a small PR for checking existance of domElement while trying to destroy eventListeners

Thank you

**Comment by ArthurTimofey** [comment]
Not sure if it will fix this for you but im just dismount the stage instead.

**Comment by s0rta** [comment]
Arthur is there a way to do that using the React Pixi stage component?

**Comment by ArthurTimofey** [comment]
> Arthur is there a way to do that using the React Pixi stage component?

Yup, you can use the key prop for the stage, set that key string in a state and update the key using set state to another value to cause re render of the canvas.
